src/__tests__/components/elements/wrapper/conditional-renderer.spec.tsx

```diff
@@ -102,10 +102,10 @@ describe("conditional-renderer", () => {
 		};
 		renderComponent(fields);
 
-		fireEvent.change(getFieldOne(), { target: { value: invalid } });
+		await waitFor(() => fireEvent.change(getFieldOne(), { target: { value: invalid } }));
```

Comment:
what is the purpose of `waitFor` for some of these events? would async `act` also work?

Reply:
the `waitFor` is added due to the change in `checkIsFormValid()` to be async, in some tests, the state is still updating after the tests have been completed.

I can use `act` as well, it will wait for the side effects to resolve though based on past experience, some times it does not wait long enough for unrelated side effects to resolve, hence I used `waitFor` here

Reply:
I guess there's no harm using it specifically in these places since it's on a per action basis